### ISSUE TITLE
[SFT] Model trainned on  Bespoke-Stratos-17k get really low performence on Math-500

### ISSUE THREAD

**Issue by ListentoMe0112** [huggingface/open-r1#396]
I'm new to llm and I'm confused by this. When I'm trainning, my loss value is decreased as expected. But at last my performance is really bad.

## Performance
<img width="510" alt="Image" src="https://github.com/user-attachments/assets/a017845a-972b-4cd9-897d-7753087319a4" />

## Device
My devices are two v100 which is not support bf16. 

## Command
This is my start trainning command
```bash
nohup accelerate launch --config_file recipes/accelerate_configs/zero3.yaml src/open_r1/sft.py --config recipes/Qwen2.5-1.5B-Instruct/sft/config_demo.yaml > train.log 2>&1 &
```

## Config
This is my config_demo.yaml

```yaml
# Model arguments
model_name_or_path: Qwen/Qwen2.5-1.5B-Instruct
model_revision: main
torch_dtype: float16
# attn_implementation: flash_attention_2

# Data training arguments
dataset_name: HuggingFaceH4/Bespoke-Stratos-17k
dataset_configs:
- all
preprocessing_num_workers: 2

# SFT trainer config
bf16: true
do_eval: true
eval_strategy: steps
eval_steps: 100
gradient_accumulation_steps: 1
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
hub_model_id: Qwen2.5-1.5B-Open-R1-Distill
hub_strategy: every_save
learning_rate: 2.0e-05
log_level: info
logging_steps: 5
logging_strategy: steps
lr_scheduler_type: cosine
packing: true
max_seq_length: 2048
max_steps: -1
num_train_epochs: 1
output_dir: data/Qwen2.5-1.5B-Open-R1-Distill
overwrite_output_dir: true
per_device_eval_batch_size: 1
per_device_train_batch_size: 1
push_to_hub: true
report_to:
- wandb
save_strategy: "no"
seed: 42
warmup_ratio: 0.1
```
## Model Card 
https://huggingface.co/yeeboxie/Qwen2.5-1.5B-Open-R1-Distill/blob/main/README.md

## Trainning Log
https://wandb.ai/1904087817-tencent/huggingface/runs/0nnfxwdj

Any help would be appreciated!

**Comment by lewtun** [comment]
Hello @ListentoMe0112 we just merged an important fix to the prompt for the math evals in #392. Could you try evaluating your model again and seeing if the resulting performance improves?

**Comment by ListentoMe0112** [comment]
> Hello [@ListentoMe0112](https://github.com/ListentoMe0112) we just merged an important fix to the prompt for the math evals in [#392](https://github.com/huggingface/open-r1/pull/392). Could you try evaluating your model again and seeing if the resulting performance improves?

Thanks for helping.
I will try it again.
But I get a few warnings as I start evaluation, is it ok?
```bash
[2025-02-23 09:49:36,408] [    INFO]: HuggingFaceH4/MATH-500 default (lighteval_task.py:187)
[2025-02-23 09:49:36,408] [ WARNING]: Careful, the task custom|math_500 is using evaluation data to build the few shot examples. (lighteval_task.py:260)
[2025-02-23 09:49:41,971] [    INFO]: --- INIT SEEDS --- (pipeline.py:259)
[2025-02-23 09:49:41,971] [    INFO]: --- RUNNING MODEL --- (pipeline.py:464)
[2025-02-23 09:49:41,971] [    INFO]: Running RequestType.GREEDY_UNTIL requests (pipeline.py:468)
[2025-02-23 09:49:42,403] [ WARNING]: You cannot select the number of dataset splits for a generative evaluation at the moment. Automatically inferring. (data.py:260)
Splits:   0%|                                                                                                                                                                                                                    | 0/1 [00:00<?, ?it/s]
[2025-02-23 09:49:42,581] [ WARNING]: context_size + max_new_tokens=33659 which is greater than self.max_length=32768. Truncating context to 0 tokens. (vllm_model.py:268)
```

**Comment by jinyangwu** [comment]
@lewtun i also found that the evaluation results after sft is 

> I'm new to llm and I'm confused by this. When I'm trainning, my loss value is decreased as expected. But at last my performance is really bad.
> 
> ## Performance
> <img alt="Image" width="510" src="https://private-user-images.githubusercontent.com/66709955/415907163-a017845a-972b-4cd9-897d-7753087319a4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAyODM4MzksIm5iZiI6MTc0MDI4MzUzOSwicGF0aCI6Ii82NjcwOTk1NS80MTU5MDcxNjMtYTAxNzg0NWEtOTcyYi00Y2Q5LTg5N2QtNzc1MzA4NzMxOWE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjIzVDA0MDUzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdlYmE3ODE1MDBjMjhmNDVkNjExYzY2NzljYmUxNmYyMmI0NjBjYjRlOWY1OTdhNjY5N2FiYjUyMDdmMWQ3ZGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xR9w-QgKl2ImaG3FUdEzY_WMiCq0UF7Ey5_Mi4HpZDc">
> ## Device
> My devices are two v100 which is not support bf16.
> 
> ## Command
> This is my start trainning command
> 
> nohup accelerate launch --config_file recipes/accelerate_configs/zero3.yaml src/open_r1/sft.py --config recipes/Qwen2.5-1.5B-Instruct/sft/config_demo.yaml > train.log 2>&1 &
> ## Config
> This is my config_demo.yaml
> 
> # Model arguments
> model_name_or_path: Qwen/Qwen2.5-1.5B-Instruct
> model_revision: main
> torch_dtype: float16
> # attn_implementation: flash_attention_2
> 
> # Data training arguments
> dataset_name: HuggingFaceH4/Bespoke-Stratos-17k
> dataset_configs:
> - all
> preprocessing_num_workers: 2
> 
> # SFT trainer config
> bf16: true
> do_eval: true
> eval_strategy: steps
> eval_steps: 100
> gradient_accumulation_steps: 1
> gradient_checkpointing: true
> gradient_checkpointing_kwargs:
>   use_reentrant: false
> hub_model_id: Qwen2.5-1.5B-Open-R1-Distill
> hub_strategy: every_save
> learning_rate: 2.0e-05
> log_level: info
> logging_steps: 5
> logging_strategy: steps
> lr_scheduler_type: cosine
> packing: true
> max_seq_length: 2048
> max_steps: -1
> num_train_epochs: 1
> output_dir: data/Qwen2.5-1.5B-Open-R1-Distill
> overwrite_output_dir: true
> per_device_eval_batch_size: 1
> per_device_train_batch_size: 1
> push_to_hub: true
> report_to:
> - wandb
> save_strategy: "no"
> seed: 42
> warmup_ratio: 0.1
> ## Model Card
> https://huggingface.co/yeeboxie/Qwen2.5-1.5B-Open-R1-Distill/blob/main/README.md
> 
> ## Trainning Log
> https://wandb.ai/1904087817-tencent/huggingface/runs/0nnfxwdj
> 
> Any help would be appreciated!


Same question! I found that after SFT on R1-distilled long-cot data, the model's responses becomes longer. However, they also include unnecessary verifications and unexpected formats. Moreover, using the lighteval codebase, the accuracy of Qwen2.5-1.5B-Instruct on MATH500 is 43.6, and SFT mdel on open-r1/OpenR1-Math-220k is 36.6. I'm really confused!

**Comment by ListentoMe0112** [comment]
@jinyangwu At least you get comparable performance, while my result is 1.6%, which is much lower than the base model.

**Comment by ListentoMe0112** [comment]
@lewtun Well, it helped a lot. I get a result of 30.8%, but it's still much lower than the base model. Thanks for your help.

**Comment by jinyangwu** [comment]
> [@jinyangwu](https://github.com/jinyangwu) At least you get comparable performance, while my result is 1.6%, which is much lower than the base model.

Well, on AIME 24, SFT model's accuracy is only 3.33% compared to base model's 43.6%. I'm figuring out which setting is causing this.

**Comment by gauss-clb** [comment]
I used LLama-factory to train qwen-2.5-7B, the accuracy will exceed 70%，but qwen-2.5-1.5B is bellow 40%.

**Comment by ListentoMe0112** [comment]
The batch size of trainning is important. To reproduce we need batch size up to 16 * 8 = 128. My 2*v100 pod is too weak to handle this.

**Comment by wanghaoyu0408** [comment]
> I used LLama-factory to train qwen-2.5-7B, the accuracy will exceed 70%，but qwen-2.5-1.5B is bellow 40%.

Hi, did you use the base model or the Instruct model?
